homeassistant/components/hangouts/hangouts_bot.py

```diff
@@ -214,10 +219,40 @@ def _on_disconnect(self):
                                                segment_type=hangouts_pb2.
                                                SEGMENT_TYPE_LINE_BREAK))
 
+        image_file = None
+        if data.get('image') is not None:
+            uri = data.get('image')
+            validate = urlparse(uri)
+            if validate.scheme:
+                try:
+                    websession = async_get_clientsession(self.hass)
+                    with async_timeout.timeout(5, loop=self.hass.loop):
+                        response = await websession.get(uri)
+                        if response.status != 200:
+                            _LOGGER.error(
+                                'Fetch image failed, {}, {}'.format(
+                                    response.status, response
+                                )
+                            )
+                            image_file = None
+                        else:
+                            image_data = await response.read()
+                            image_file = io.BytesIO(image_data)
+                            image_file.name = "image.png"
+                except (asyncio.TimeoutError, aiohttp.ClientError) as error:
+                    _LOGGER.error(
+                        'Failed to fetch image, {}'.format(
+                            type(error)
+                        )
+                    )
+                    image_file = None
+            else:
+                image_file = open(uri, 'rb')
```

Comment:
You need check the path of file with our security function: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/core.py#L1183